File: parking-lot/java/VehicleType.java
public enum VehicleType {
    MOTORCYCLE("Motorcycle"),
    CAR("Car"),
    TRUCK("Truck"),
    UNKNOWN("Unknown");

    private final String name;

    VehicleType(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: parking-lot/go-without-multithreading/enums/vehicle_type.go
package enums

type VehicleType int

const (
	Motorcycle VehicleType = iota
	Car
	Truck
)

func (v VehicleType) String() string {
	switch v {
	case Motorcycle:
		return "Motorcycle"
	case Car:
		return "Car"
	case Truck:
		return "Truck"
	default:
		return "Unknown"
	}
}
